testnet/stacks-signer/src/net.rs

```diff
@@ -0,0 +1,62 @@
+use crate::config::Config;
+use crate::signer;
+use serde::Serialize;
+use std::fmt::{Debug, Formatter};
+use ureq::Response;
+
+pub struct HttpNet {
+    _highwater_msg_idx: usize,
+    stacks_node_url: String,
+}
+pub trait Net {
+    fn listen(&self);
+    fn poll(&self) -> Result<Response, ureq::Error>;
+    fn next_message(&self) -> Message;
+    fn send_message(&self, _msg: Message);
+}
+
+#[derive(Debug)]
+pub struct Message {
+    pub r#type: signer::MessageTypes,
```

Comment:
Style nit: please use variable names that don't collide with reserved keywords.  Thanks!